connectionmgr.go

```diff
@@ -0,0 +1,72 @@
+package utils
+
+import (
+	"log"
+	"sync"
+)
+
+// Initializer provides an interface an object to provide methods for startup and shutdown.
+type Initializer interface {
+	Initialize()
+	Close()
+}
+
+// CloseFunc is a function that gracefully shuts down a connection as a side effect.
+type CloseFunc func()
+
+// ConnectionManager manages initialization and shutdown of long lived connections.
+// Each connection object must match the Initializer interface. Initialization happens
+// in parallel. The waitgroup can be used to wait util all connections are initialized.
+// Close kick off the proper shutdown of all managed connections.
+type ConnectionManager struct {
+	closerChan chan CloseFunc
+	Close      CloseFunc
+	WG         sync.WaitGroup
+}
+
+// NewConnectionManager initializes a new ConnectionManager object that can be used
+// to manage the life of long lived remote connections such as to a database.
```

Comment:
Did you look at/think about using context for this purpose? https://blog.golang.org/context